src/firmware/app/primitives/estop_primitive.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "firmware/app/primitives/primitive.h"
+#include "shared/proto/primitive.nanopb.h"
+
+extern const primitive_t ESTOP_PRIMITIVE;
+
+/**
+ * Start the Estop primitive in the given world
+ *
+ * @param prim_msg The msg for the primitive
+ * @param void_state_ptr [in] A pointer to this primitives state, as allocated by the
+ *                            `create_state` function
+ * @param world [in] The world this primitive is running in.
```

Comment:
Checking if this needs `[in]` if it's a "pseudo-class"

Reply:
yeah this is a weird case where `in` is useful because it says that the function won't modify those arguments. I don't think it hurts to have it

Reply:
Yeah, this is a bit of a tricky one. One could say you're modifying the world in the primitive, but we're not actually modifying the data structure, we're just using the functions inside the data structure. Anyhow, might be worth doc'ing when to use `[in]`, `[out]`, and `[in/out]` in the docs.

Reply:
this is the code style convention:
* Functions that return values via argument(s) must have all **pointer** parameters labelled as `[in]`, `[in/out]`, or `[out]` in the javadoc, **in that order**. The exceptions to this are [pseudo-class](/docs/firmware-architecture-and-design.md#pseudo-class) functions, which should take the "class" (which is an `[in/out]` as the first argument) and function pointers.

So `world` and `primitive` are pointers, but having [in] is a helpful comment so not worth removing imo